projects/canopy/src/lib/forms/input/input-field.component.ts

```diff
@@ -47,28 +80,108 @@ export class LgInputFieldComponent {
     this._inputElement = element;
     this._inputElement.id = this.id;
   }
+  get inputElement() {
+    return this._inputElement;
+  }
 
-  _hintElement: LgHintComponent;
+  private _hintElement: LgHintComponent;
   @ContentChild(LgHintComponent)
   set hintElement(element: LgHintComponent) {
-    this._inputElement.ariaDescribedBy = this.domService.toggleIdInStringProperty(
-      this._inputElement.ariaDescribedBy,
+    this.inputElement.ariaDescribedBy = this.domService.toggleIdInStringProperty(
+      this.inputElement.ariaDescribedBy,
       this._hintElement,
       element,
     );
     this._hintElement = element;
   }
 
-  _validationElement: LgValidationComponent;
+  private _validationElement: LgValidationComponent;
   @ContentChild(LgValidationComponent)
   set errorElement(element: LgValidationComponent) {
-    this._inputElement.ariaDescribedBy = this.domService.toggleIdInStringProperty(
-      this._inputElement.ariaDescribedBy,
+    this.inputElement.ariaDescribedBy = this.domService.toggleIdInStringProperty(
+      this.inputElement.ariaDescribedBy,
       this._validationElement,
       element,
     );
     this._validationElement = element;
   }
 
+  @ContentChild(LgButtonComponent) buttonElement: LgButtonComponent;
+
+  private _suffixChildren: QueryList<LgSuffixDirective>;
+  @ContentChildren(LgSuffixDirective)
+  set suffixChildren(elements: QueryList<LgSuffixDirective>) {
+    elements.forEach((element) => {
+      this.inputElement.ariaDescribedBy = this.domService.toggleIdInStringProperty(
+        this.inputElement.ariaDescribedBy,
+        this._validationElement,
+        element,
+      );
+    });
+    this._suffixChildren = elements;
+  }
+  get suffixChildren() {
+    return this._suffixChildren;
+  }
+
+  private _prefixChildren: QueryList<LgSuffixDirective>;
+  @ContentChildren(LgPrefixDirective)
+  set prefixChildren(elements: QueryList<LgSuffixDirective>) {
+    elements.forEach((element) => {
+      this.inputElement.ariaDescribedBy = this.domService.toggleIdInStringProperty(
+        this.inputElement.ariaDescribedBy,
+        this._validationElement,
+        element,
+      );
+    });
+    this._prefixChildren = elements;
+  }
+  get prefixChildren() {
+    return this._prefixChildren;
+  }
+
+  @Input() id = `lg-input-${nextUniqueId++}`;
+
   constructor(private domService: LgDomService) {}
+
+  /*
+    The input field control element mimics the border of the input field.
+    This allows us to add buttons and icons inside the input field.
+    Lack of IE11 support for :focus-within necessitates us doing this in JS
+  */
+  private hasFocus = false;
+  private hasHover = false;
+  private disabledStateChanges: Subscription;
+
+  ngAfterContentInit() {
```

Comment:
I would like us to start using return types a bit more. Could you please add the return types for these functions?

Reply:
Good idea, I've added a bunch to this file, I wonder if there is a lint warning we can add?

Reply:
Yes I'm sure there is as we have it in our project. Might be one for when we create the global linting?